src/qibocal/auto/task.py

```diff
@@ -74,28 +67,34 @@ def priority(self):
 
     @property
     def parameters(self):
-        return Parameters.load(self.action.parameters)
+        return self.operation.parameters_type.load(self.action.parameters)
 
     def datapath(self, base_dir: Path):
-        return base_dir / f"{self.id}_{self.iteration}" / DATAFILE
+        path = base_dir / "data" / f"{self.id}_{self.iteration}"
+        os.makedirs(path)
+        return path
 
     def data(self, base_dir) -> Optional[Data]:
-        if not self.datapath(base_dir).is_file():
+        if not self.datapath(Path(base_dir)).is_file():
             return None
-
         Data = self.operation.data_type
+        # print(self.datapath(Path(base_dir)))
+        return Data.load_data(
+            base_dir, "data", f"{self.id}_{self.iteration}", "csv", "data"
+        )
+        # return Data(yaml.safe_load(self.datapath(base_dir).read_text(encoding="utf-8")))
 
-        return Data(yaml.safe_load(self.datapath(base_dir).read_text(encoding="utf-8")))
-
-    def run(self) -> Results:
+    def run(self, platform, qubits, folder) -> Results:
         try:
             operation: Routine = self.operation
             parameters = self.parameters
         except RuntimeError:
             operation = dummy_operation
             parameters = DummyPars()
 
-        data: Data = operation.acquisition(parameters)
+        path = self.datapath(Path(folder))
+        data: Data = operation.acquisition(platform, qubits, parameters)
+        data.to_csv(path)
```

Comment:
```suggestion
        data: Data = operation.acquisition(platform, qubits, parameters)
        data.to_csv(self.datapath(folder))
```

As above, we can assume a `Path` to be passed. Moreover, if you use it only once you don't need a name, unless:
- your name is specific enough, so it describes better the returned object (but `path` is rather generic)
- it would result in a very long call (but this is not the case)